back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/util/ErrorsMap.java

```diff
@@ -0,0 +1,102 @@
+package com.ubikloadpack.jmeter.ulp.observability.util;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+/**
+ * The ErrorsMap stores for each error type it's occurrences.
+ * The occurrences of an error type is handled by {@link ErrorTypeInfo}
+ *
+ */
+public class ErrorsMap {
+	private ConcurrentHashMap<String, ErrorTypeInfo> errorsPerType;
+	
+	public ErrorsMap() {
+		this.errorsPerType = new ConcurrentHashMap<>();
+	}
+
+	/**
+	 * Add an error type and increment it's occurrence.
+	 * @param errorType the error type to add.
+	 * @return The information of the added error type.
+	 */
+	public ErrorTypeInfo addErrorTypeAndCount(String errorType) {
+		ErrorTypeInfo errorTypeInfo = this.errorsPerType.computeIfAbsent(errorType, s -> new ErrorTypeInfo(errorType));
+	    errorTypeInfo.increment();
+	    return errorTypeInfo;
+	}
+	
+	/**
+	 * Compute the frequency of that error type.
+	 * @param errorType the type of error for which the frequency will be calculated
+	 * @param totalErrors the total errors. Should get it's value from the Micrometer registry.
+	 * @return Percentage of occurrences of a specific error type out of the total number of errors.
+	 * Returns null if no errorType key was found.
+	 */
+	public Double computeErrorTypeFrequency(String errorType, Long totalErrors) {
+		if (this.errorsPerType.contains(errorType)) {

```

Comment:
Why not proceed like addErrorTypeAndCount?
Otherwise, instead of returning null, return Optional<Double> although it looks to me as abnormal case

Reply:
i don't understand how can i proceed like addErrorTypeAndCount